stratum/hal/lib/common/switch_interface.h

```diff
@@ -150,25 +149,33 @@ class SwitchInterface {
       WriterInterface<::p4::v1::ReadResponse>* writer,
       std::vector<::util::Status>* details) = 0;
 
-  // Registers a writer to be invoked when we receive a packet on any port on
-  // the specified node which are destined for the controller. The sent
-  // ::p4::PacketIn instance includes all the info on where the packet was
-  // received on this node as well as its payload.
-  virtual ::util::Status RegisterPacketReceiveWriter(
+  // Registers a writer to be invoked when we receive a StreamMessageResponse on
+  // the specified node which are destined for the controller. A
+  // StreamMessageResponse can carry many different types of sub-messages, such
+  // as PacketIns, digests or idle timeout notifications.
+  // PacketIns: The sent ::p4::PacketIn instance includes all the info on
+  //      where the packet was received on this node as well as its payload.
+  virtual ::util::Status RegisterStreamMessageResponseWriter(
       uint64 node_id,
-      std::shared_ptr<WriterInterface<::p4::v1::PacketIn>> writer) = 0;
+      std::shared_ptr<WriterInterface<::p4::v1::StreamMessageResponse>>
```

Comment:
Can we make this a const reference instead of a copy?

Reply:
Here is the literature I found on this topic:

https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs
> Avoid defining functions that require a const reference parameter to outlive the call, because const reference parameters bind to temporaries.

https://abseil.io/tips/116
> It is still OK to pass an argument by const reference to a constructor if the argument is copied, or just used in the constructor and no reference to it is kept in the constructed object.

https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/
> Guideline: Express that a function will store and share ownership of a heap object using a by-value shared_ptr parameter.

https://stackoverflow.com/questions/3310737
> Shortly, there is no reason to pass by value, unless the goal is to share ownership of an object